src/validators/register.rs

```diff
@@ -0,0 +1,99 @@
+use regex::Regex;
+
+pub fn validate_username(username: String) -> (String, String) {
+    if username.len() < 3 {
+        return (
+            username,
+            "Username must be at least 3 characters long".to_string(),
+        );
+    }
+    if username.len() > 20 {
+        return (
+            username,
+            "Username must be at most 20 characters long".to_string(),
+        );
+    }
+    if !username
+        .chars()
+        .all(|c| c.is_ascii_alphanumeric() || c == '_')
+    {
+        return (
+            username,
+            "Username must contain only alphanumeric characters and underscores".to_string(),
+        );
+    }
+
+    //TODO: check if username is already taken
+    (username, String::new())
+}
+
+pub fn vlaidate_name(name: String) -> (String, String) {
+    if name.len() < 2 {
+        return (name, "Name must be at least 2 characters long".to_string());
+    }
+    (name, String::new())
+}
+
+pub fn validate_email(email: String) -> (String, String) {
+    let email_regex = Regex::new(r"(?i)^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$").unwrap();
```

Comment:
I believe this regex is correct. Not gonna check xD

Reply:
Copilot generated